code/common/grafana.q

```diff
@@ -65,23 +73,41 @@ diskvals:{c:(count[x]-ticks)+til ticks;get'[.Q.ind[x;c]]};
 memvals:{get'[?[x;enlist(within;`i;count[x]-ticks,0);0b;()]]};
 catchvals:{@[diskvals;x;{[x;y]memvals x}[x]]};
 
+istype:{[targ;char] (char,del)~2#targ};
+isfunc:istype[;"f"]; istab:istype[;"t"];
+
+// builds body of table response in Json adaptor schema
+tabresponse:{[colname;coltype;rqt] .j.j enlist`columns`rows`type!(flip`text`type!(colname;coltype);catchvals rqt;`table)};
+
 // process a table request and return in JSON format
 tbfunc:{[rqt]
-  rqt:value raze rqt[`targets]`target;
+  rqt: raze rqt[`targets]`target;
+  symname:0b;
+  // if f.t.func, drop first 4 chars
+  $[isfunc[rqt] & istab[2_rqt]; rqt:4_rqt; 
```

Comment:
if else $ returns a value so you could use `rqt:[isfunc[rqt] & istab[2_rqt]; 4_rqt; ...` instead